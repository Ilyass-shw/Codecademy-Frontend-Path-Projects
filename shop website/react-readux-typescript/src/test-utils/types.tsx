import { configureStore, createSlice } from '@reduxjs/toolkit';
import { render } from '@testing-library/react';


export type StoreType = ReturnType<typeof configureStore>;
export type RenderReturn = ReturnType<typeof render>;
export type SliceParameters = Parameters<typeof createSlice>;
/* eslint-disable */
export type UiType = React.ReactElement<
  any,
  string | React.JSXElementConstructor<any>
>;
/* eslint-enable */

export interface TestProviderprops {
  MyStore: StoreType;
  children: /* eslint-disable */
  React.ReactElement<any, string | React.JSXElementConstructor<any>>;
  /* eslint-enable */
}
