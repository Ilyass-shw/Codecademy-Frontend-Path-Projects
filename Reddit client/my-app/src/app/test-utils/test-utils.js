import React from "react";
import { render as rtlRender } from "@testing-library/react";
// import { createStore } from "redux";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

const mockStore = configureStore([]);

let Store;
Store = mockStore({});

function render(ui, { initialState, store = Store, ...renderOptions } = {}) {
	function Wrapper({ children }) {
		return <Provider store={store}>{children}</Provider>;
	}
	return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything
export * from "@testing-library/react";
// override render method
export { render };
