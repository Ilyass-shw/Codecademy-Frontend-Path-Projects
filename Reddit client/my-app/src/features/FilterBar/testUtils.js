import React from 'react'
import { render as rtlRender } from '@testing-library/react'
import { Provider } from 'react-redux'
import store from '../../app/store'

const TestProvider = ({
  store,
  children
}) => <Provider store={store}>{children}</Provider>

export function render(ui, { store, ...otherOpts }) {
  return rtlRender(<TestProvider store={store}>{ui}</TestProvider>, otherOpts)
}

export function makeTestStore(opts = {}) {
  const origDispatch = store.dispatch
  store.dispatch = jest.fn(origDispatch)
  return store
}

// re-export everything
export * from "@testing-library/react";

