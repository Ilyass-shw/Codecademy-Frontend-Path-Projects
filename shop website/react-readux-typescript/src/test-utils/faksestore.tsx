// import { configureStore } from '@reduxjs/toolkit';
// import { createSlice } from '@reduxjs/toolkit';

// import { StoreType , SliceParameters} from './types';

// // const initialState :object= {};
// const fakeSlice = createSlice({
//   name: 'posts',
//   initialState,
//   reducers: {},
// });

// export default configureStore({
//   reducer: {
//     posts: fakeSlice.reducer,
//   },
// });

// export const makeCustomTestStore = (
//   name: string,
//   initialState: string[],
//   reducers = {},
// ): StoreType => {
//   const fakeSlice = createSlice({
//     name,
//     initialState,
//     reducers,
//   });

//   return configureStore({
//     reducer: {
//       posts: fakeSlice.reducer,
//     },
//   });
// };
