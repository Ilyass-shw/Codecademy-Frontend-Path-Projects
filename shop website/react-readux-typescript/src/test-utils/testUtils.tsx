import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { Provider } from 'react-redux';
import store, { makeCustomTestStore } from './faksestore';
import { RenderReturn, StoreType } from './types';
import { BrowserRouter } from 'react-router-dom';

// const TestWrapper: React.FC<WrapperParams> = ({
//   MyStore,
//   WithRouter,
//   children,
// }) => {
//   if (WithRouter) {
//     return (
//       <Provider store={MyStore}>
//         <BrowserRouter>{children}</BrowserRouter>
//       </Provider>
//     );
//   } else {
//     return <Provider store={MyStore}>{children}</Provider>;
//   }
// };

// ============= ============= ============= ============= =============

export const render = (
  ui: React.ReactElement,
  MyStore: StoreType,
  WithRouter?: 'withRouter',
): RenderReturn => {
  if (WithRouter) {
    return rtlRender(
      <Provider store={MyStore}>
        <BrowserRouter>{ui}</BrowserRouter>
      </Provider>,
    );
  } else {
    return rtlRender(<Provider store={MyStore}>{ui}</Provider>);
  }
};

// ============= ============= ============= ============= =============

export const renderWithRouterOnly = (ui: React.ReactElement): RenderReturn => {
  return rtlRender(<BrowserRouter>{ui}</BrowserRouter>);
};

// ============= ============= ============= ============= =============

export const makeTestStore = (
  Name?: string,
  initialState?: Record<string, unknown>,
  reducers = {},
): StoreType => {
  if (Name && initialState) {
    const CustomTestStore = makeCustomTestStore(Name, initialState, reducers);
    const origDispatch = CustomTestStore.dispatch;
    CustomTestStore.dispatch = jest.fn(origDispatch);
    return CustomTestStore;
  } else {
    const origDispatch = store.dispatch as jest.Mock;
    store.dispatch = jest.fn(origDispatch);
    return store;
  }
};

// re-export everything
export * from '@testing-library/react';
