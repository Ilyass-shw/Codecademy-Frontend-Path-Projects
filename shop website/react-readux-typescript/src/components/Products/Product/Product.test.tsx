import React from 'react'
import { render } from '../../../test-utils/testUtils'

describe('Product',()=>{
    it('should render', () => {
        render()
    })
    
})