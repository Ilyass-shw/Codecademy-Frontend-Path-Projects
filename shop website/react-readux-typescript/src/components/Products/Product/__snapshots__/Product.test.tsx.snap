// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product should match snaptshot 1`] = `
<DocumentFragment>
  <li
    class="sc-gKckTs jFUhbv"
  >
    <a
      class="sc-eCImvq sc-iCfLBT gECYPg bNwGWt"
      href="/product/1"
    >
      <div>
        <div
          class="sc-bdvvaa cYloRB"
        >
          <img
            alt="itemOne"
            class="sc-gsDJrp dlZrhb"
            loading="lazy"
            src="src1"
          />
          <img
            alt="itemOne"
            class="sc-gsDJrp dmjAPj"
            loading="lazy"
            src="src2"
          />
          <img
            alt="itemOne"
            class="sc-gsDJrp fGGvVI"
            loading="lazy"
            src="src3"
          />
        </div>
        <div
          class="sc-dkPtyc goXLUj"
        >
          <div
            class="sc-hKwCoD kLaVan"
          />
        </div>
      </div>
      <h4
        class="sc-furvIG boMFUg"
      >
        itemOne
      </h4>
      <h6
        class="sc-pVTma bEbIhi"
      >
        $45
      </h6>
    </a>
  </li>
</DocumentFragment>
`;
